### PR TITLE
Documentation for rebasing pull requests

### PR DESCRIPTION
**Pull request by dst**

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -12,3 +12,17 @@
 * Be carefull when changing dependencies
 * You don't have to assign person and milestone
 * Document new features in [Wiki](https://github.com/4finance/micro-infra-spring/wiki), especially new [configuration properties](https://github.com/4finance/micro-infra-spring/wiki/Configuration) after they are merged
+
+# Rebasing your pull request
+You can consider rebasing your pull request instead of merging it. Let's assume that you have
```
Comment (nurkiewicz) [inline CONTRIBUTING.md]: "You should prefer"

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -12,3 +12,17 @@
...
+a branch named cool-feature. Here are instructions needed to rebase your branch on top of master:
```
Comment (nurkiewicz) [inline CONTRIBUTING.md]: Use quotes: `cool-feature`.

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -12,3 +12,17 @@
...
+
+        $ git pull
```
Comment (nurkiewicz) [inline CONTRIBUTING.md]: First do `git checkout master`

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -12,3 +12,17 @@
...
+        $ git checkout cool-feature
+        $ git rebase master
```
Comment (nurkiewicz) [inline CONTRIBUTING.md]: "Resolve commits if any"

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -12,3 +12,17 @@
...
+        $ git push origin cool-feature -f
+        $ git checkout master
```
Comment (nurkiewicz) [inline CONTRIBUTING.md]: Before pushing to master you ideally wait for Travis/Snap CI results, making sure your code isn't broken after rebase.

**Comment by szpak** [comment]
Maybe we could give recommendation in doc to make a pull request to GitHub to allow to do that from a button (maybe with sample content to paste into https://github.com/contact)?

File: CONTRIBUTING.md (hunk 4 above)
Comment (marcingrzejszczak) [inline CONTRIBUTING.md]: shouldn't it be:

```
git rebase origin/master
```
?

Comment (szpak) [inline CONTRIBUTING.md]: After `git pull` the local master should be up-to-date, but rebasing with origin/master also could be fine.

**Comment by dst** [comment]
@szpak, very good idea.
Could you add a sample content? I will use it as well :)

**Comment by dst** [comment]
@nurkiewicz, thanks for comments. Everyhing is fixed in second commit.

**Pull request merged**